core/src/main/java/org/mule/runtime/core/api/context/DefaultMuleContextBuilder.java

```diff
@@ -106,9 +108,13 @@ public MuleContext buildMuleContext() {
         .setBootstrapServiceDiscoverer(injectMuleContextIfRequired(getBootstrapPropertiesServiceDiscoverer(), muleContext));
 
     getObjectSerializer(muleContext);
-    ErrorTypeRepository defaultErrorTypeRepository = createDefaultErrorTypeRepository();
-    muleContext.setErrorTypeRepository(defaultErrorTypeRepository);
-    muleContext.setErrorTypeLocator(createDefaultErrorTypeLocator(defaultErrorTypeRepository));
+
+    if (errorTypeRepository == null) {
+      errorTypeRepository = createDefaultErrorTypeRepository();
+    }
```

Comment:
I think you are missing the else here where you create it passing along the existing one.